.github/workflows/master.yml

```diff
@@ -38,19 +50,74 @@ jobs:
           token: ${{ secrets.GITHUB_TOKEN }}
           args: --all-features -- -D warnings
 
-  solana-build: 
+      - name: Miri tests
+        run: cargo miri test -- -Z unstable-options --report-time --skip ::anchor
+
+  anchor-build:
+    name: Anchor Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Install Rust
+        id: install-rust
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: stable
+          components: rustfmt
+
+      - name: Cache Anchor
+        id: cache-anchor
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.avm
+            ~/.cargo/bin/avm
+            ~/.cargo/bin/anchor
+            ~/.config/solana/
+            ~/.local/share/solana/
+          key: ${{ runner.os }}-${{ steps.install-rust.outputs.cachekey }}-${{ env.SOLANA_VERSION }}-${{ env.ANCHOR_VERSION }}-anchor
+
+      - name: Install Solana
+        if: steps.cache-anchor.outputs.cache-hit != 'true'
+        run: |
+          set -eux
+          curl -sSfL https://release.solana.com/$SOLANA_VERSION/install | sh
+          cp solana/solana-ibc/keypair.json ~/.config/solana/id.json
+
+      - name: Install Solana PATH
+        run: echo "$HOME/.local/share/solana/install/active_release/bin" >> $GITHUB_PATH
+
+      - name: Install Anchor
+        if: steps.cache-anchor.outputs.cache-hit != 'true'
+        run: |
+          set -eux
+          cargo install --git https://github.com/coral-xyz/anchor avm --locked --force
+          avm install $ANCHOR_VERSION
+          avm use $ANCHOR_VERSION
+
+      - name: Anchor Build
+        run: anchor build
+
+      - name: Anchor Build (with mocks)
+        run: anchor build -- --features=mocks
+
+  solana-build:
```

Comment:
I dont think we have to build we need this, because `anchor build` runs `cargo build-sbf` under the hood anyways.

Reply:
I wondered about that. What stopped me though is that `anchor build` takes a bit over five minutes while `cargo build-sbf` takes a bit over six minutes. Though it’s possible that anchor is smarter about what it builds with sbf I suppose.

Reply:
I thinks anchor optimizes somehow. https://github.com/coral-xyz/anchor/blob/master/cli/src/config.rs#L425